### PR TITLE
Converters: add missing tests and add linting to converter tests

### PR DESCRIPTION
**Pull request by bernt-matthias**
While working on https://github.com/galaxyproject/galaxy/pull/12073 I realized that tests of many converters use the wrong attribute `format` instead of `ftype` and that tests are missing for many converters.

The PR

- adds `planemo lint` to the converter tests
- adds tests to almost all converters (a few are excluded via a new skiplist)

This replaces https://github.com/galaxyproject/galaxy/pull/12104 (in parts).

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/datatypes/converters/bz2_to_uncompressed.xml [hunk 1]
```diff
@@ -1,23 +1,34 @@
...
-  </inputs>
-  <outputs>
-      <!-- auto doesn't sniff - it defers to galaxy.json in this context. -->
-      <data name="output1" format="auto"/>
-  </outputs>
+}}
+    </configfile>
+    </configfiles>
+    <inputs>
+        <param format="data" name="input1" type="data" label="Choose compressed file"/>
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/converters/bz2_to_uncompressed.xml]:
```suggestion
        <param name="input1" type="data" format="data" label="Choose compressed file"/>
```

File: lib/galaxy/datatypes/converters/bz2_to_uncompressed.xml [hunk 2]
```diff
@@ -1,23 +1,34 @@
...
-  </configfiles>
-  <inputs>
-      <param format="data" name="input1" type="data" label="Choose compressed file"/>
-  </inputs>
-  <outputs>
-      <!-- auto doesn't sniff - it defers to galaxy.json in this context. -->
-      <data name="output1" format="auto"/>
-  </outputs>
+}}
+    </configfile>
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/converters/bz2_to_uncompressed.xml]:
```suggestion
}}</configfile>
```

File: lib/galaxy/datatypes/converters/bz2_to_uncompressed.xml [hunk 3]
```diff
@@ -1,23 +1,34 @@
...
+        <!-- auto doesn't sniff - it defers to galaxy.json in this context. -->
+        <data name="output1" format="auto"/>
+    </outputs>
+    <tests>
+        <test>
+            <param name="input1" value="1.fastqsanger.bz2"/>
+            <output name="output1" ftype="fastqsanger">
+              <assert_contents>
+                <has_text text="@1831_573_1004/1"/>
+              </assert_contents>
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/converters/bz2_to_uncompressed.xml]:
```suggestion
                <assert_contents>
                    <has_text text="@1831_573_1004/1"/>
                </assert_contents>
```

File: lib/galaxy/datatypes/converters/bz2_to_uncompressed.xml [hunk 4]
```diff
@@ -1,23 +1,34 @@
...
+            </output>
+        </test>
+    </tests>
   <help>
   </help>
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/converters/bz2_to_uncompressed.xml]:
```suggestion
    <help>
    </help>
```

File: lib/galaxy/datatypes/converters/cram_to_bam_converter.xml [hunk 5]
```diff
@@ -2,13 +2,23 @@
     <requirements>
         <requirement type="package" version="0.15.4">pysam</requirement>
     </requirements>
-    <command><![CDATA[python $__tool_directory__/cram_to_bam.py '$input' '$output']]></command>
+    <command><![CDATA[python '$__tool_directory__/cram_to_bam.py' '$input' '$output']]></command>
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/converters/cram_to_bam_converter.xml]:
```suggestion
    <command><![CDATA[
python '$__tool_directory__/cram_to_bam.py' '$input' '$output'
    ]]></command>
```

File: lib/galaxy/datatypes/converters/encodepeak_to_bgzip_converter.xml [hunk 6]
```diff
@@ -12,11 +12,17 @@
         '$input1' '$output1'
     </command>
     <inputs>
-        <param format="ENCODEPeak" name="input1" type="data" label="Choose ENCODEPeak file"/>
+        <param format="encodepeak" name="input1" type="data" label="Choose ENCODEPeak file"/>
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/converters/encodepeak_to_bgzip_converter.xml]:
```suggestion
        <param name="input1" type="data" format="encodepeak" label="Choose ENCODEPeak file"/>
```

**Comment by bernt-matthias** [comment]
Heureka. Tests are passing. Good to go from my side.

**Comment by jmchilton** [comment]
Amazing - thank you so much!

**Pull request merged**